helix-core/src/syntax.rs

```diff
@@ -1830,6 +1832,7 @@ fn injection_for_match<'a>(
 pub struct Merge<I> {
     iter: I,
     spans: Box<dyn Iterator<Item = (usize, std::ops::Range<usize>)>>,
+    emit_selection_events: bool,
```

Comment:
I'm not sure this would actually be necessary for future merges/iterators but I think it would be more elegant to have this be

```rs
pub struct Merge<I> {
    // ..
    events: Option<(HighlightEvent, HighlightEvent)>,
    //                  ^ start event     ^ end event
    // ..
}
```

passing `None` for merges like diagnostics where we don't need to emit events and `Some((SelectionStart, SelectionEnd))` for selections when selection whitespace rendering is enabled.

The boolean just for selection events seems very specific for this otherwise very generic merge function/type.

Reply:
Thanks for your review.  I've rebased and implemented your suggestion in the latest revision.